jwt/jwk.py

```
@@ -279,25 +276,139 @@ def jwk_from_dict(dct: Mapping[str, Any]) -> AbstractJWKBase:
+
+    def wrapper(content, **kwargs):
+        # content is the only positional argument allowed
+        # the function you wrap must use kwargs-only arguments PEP3102
+        if not isinstance(content, bytes):  # pragma: no cover
+            raise TypeError(
+                'content must be bytes, it is {}'.format(type(content)))
+        load_function = f'serializer_load_function_{kind}'
+        if kwargs.get(load_function) is None:
+            raise Exception(f'{load_function} is not Optional')
+        # now convert it to a Callable if it's a string
+        if isinstance(kwargs.get(load_function), str):
+            kwargs[load_function] = getattr(
+                serialization_module,
+                kwargs.get(load_function),
+            )
+
+        if kwargs.get('options') is None:
+            kwargs['options'] = {}
+
+        return func(content, **kwargs)
+    return wrapper
+
+
+@jwk_from_bytes_argument_conversion
+def jwk_from_private_bytes(
+    content: bytes,
+    *,
+    serializer_load_function_private:
+        Union[str, Callable[[bytes, Optional[str], object], object]] = None,
+    password: Optional[str] = None,
+    backend: Optional[object] = None,
+    options: Optional[dict] = None,
+) -> Optional[AbstractJWKBase]:
+    """This function is meant to be called from jwk_from_bytes"""
     try:
-        privkey = load_pem_private_key(
-            pem_content, password=None, backend=default_backend())
+        privkey = serializer_load_function_private(
+            content,
+            password=password,
+            backend=backend,
+        )
         if isinstance(privkey, RSAPrivateKey):
-            return RSAJWK(privkey)
-        raise UnsupportedKeyTypeError(
-            'unsupported key type')  # pragma: no cover
+            return RSAJWK(privkey, **options)
+        raise UnsupportedKeyTypeError('unsupported key type')
     except ValueError:
-        pass
-
+        return None
```

Returning None is not consistent behavior with `jwk_from_public_bytes` which raises UnsupportedKeyTypeError. So I think it is better to raise `UnsupportedKeyTypeError` here and fallback to `jwk_from_public_bytes`.

---

you're right, for the exception message is 'this is probably a public key' good?

---

Yes, thanks!